src/Parsers/New/CMakeLists.txt

```diff
@@ -0,0 +1,97 @@
+set (SRCS
+    AST/AlterTableQuery.cpp
+    AST/AttachQuery.cpp
+    AST/CheckQuery.cpp
+    AST/ColumnExpr.cpp
+    AST/ColumnTypeExpr.cpp
+    AST/CreateDatabaseQuery.cpp
+    AST/CreateDictionaryQuery.cpp
+    AST/CreateLiveViewQuery.cpp
+    AST/CreateMaterializedViewQuery.cpp
+    AST/CreateTableQuery.cpp
+    AST/CreateViewQuery.cpp
+    AST/DDLQuery.cpp
+    AST/DescribeQuery.cpp
+    AST/DropQuery.cpp
+    AST/EngineExpr.cpp
+    AST/ExistsQuery.cpp
+    AST/ExplainQuery.cpp
+    AST/Identifier.cpp
+    AST/InsertQuery.cpp
+    AST/JoinExpr.cpp
+    AST/KillQuery.cpp
+    AST/LimitExpr.cpp
+    AST/Literal.cpp
+    AST/OptimizeQuery.cpp
+    AST/OrderExpr.cpp
+    AST/Query.cpp
+    AST/RatioExpr.cpp
+    AST/RenameQuery.cpp
+    AST/SelectUnionQuery.cpp
+    AST/SetQuery.cpp
+    AST/SettingExpr.cpp
+    AST/ShowCreateQuery.cpp
+    AST/ShowQuery.cpp
+    AST/SystemQuery.cpp
+    AST/TableElementExpr.cpp
+    AST/TableExpr.cpp
+    AST/TruncateQuery.cpp
+    AST/UseQuery.cpp
+    AST/WatchQuery.cpp
+    CharInputStream.cpp
+    ClickHouseLexer.cpp
+    ClickHouseParser.cpp
+    ClickHouseParserVisitor.cpp
+    LexerErrorListener.cpp
+    parseQuery.cpp
+    ParserErrorListener.cpp
+    ParseTreeVisitor.cpp
+)
+
+add_library (clickhouse_parsers_new ${SRCS})
+
+# XXX: hack for old clang-10!
+check_cxx_compiler_flag("-Wsuggest-destructor-override" HAS_SUGGEST_DESTRUCTOR_OVERRIDE)
+
+# XXX: hack for old gcc-10!
+check_cxx_compiler_flag("-Wshadow" HAS_SHADOW)
+
+target_compile_options (clickhouse_parsers_new
+    PRIVATE
+        -Wno-c++20-compat
```

Comment:
I use clang 11, it doesn't support this parameter, using `-Wno-c++2a-compat` instead.

```
Apple clang version 11.0.3 (clang-1103.0.32.29)
Target: x86_64-apple-darwin19.6.0
Thread model: posix
InstalledDir: /Library/Developer/CommandLineTools/usr/bin
```

Reply:
We don't support build with native Apple Clang compiler. Even if it works now, it may accidentally break in any PR since we doesn't test it in CI. It's better to use Clang from LLVM.